### ISSUE TITLE
Step View: Many redundant http requests make it slow

### ISSUE THREAD

**Issue by bruderol** [scenarioo/scenarioo#544]
The Step View loads many data on initialisation, that is not specific for the selected step, which makes browsing from step to step very slow.

Same problem might also be there in other views, but for step view it is most bad.

I analysed it and found following calls, that should be done by the app context (only when needed) and should be consumed by the step view from the app context to avoid reloading this data on every step:
* labelConfigurations
* branches (3 times)
* buildDiffInfo

Especially loading the branches info so often for each of our views should be highly avoided!

Since we do not have HTTP Caching Headers well configured for our REST calls in the backend, this calls are also not cached by the browser, that's why it is even worse.

**Comment by danielsuter** [comment]
If I understand it correctly, even with caching we would make a request. We're using ETAGS. So the client will ask the server if something has changed providing the etag, if nothing has changed a 304 will be returned. So no data is transferred but there was a request nevertheless.

**Comment by bruderol** [comment]
Are you sure, that we set the ETAG correctly? That was the plan since the beginning, but I make a bet that those have never been set correctly. I guess that either a call is allways cached (even if the resource changes) or never cached. I bet for the second, but who knows. Can you check?

**Comment by bruderol** [comment]
Even if it is cached the 3 times loading the branches is still not nice, because it still means to ask the server for changes to the ETAG which is a needless roundtrip in that case.

**Comment by danielsuter** [comment]
I'm pretty sure @mi-we investigated that in our mini camp and came to this conclusion.

Your second comment is actually my point, that server side caching does not fix the issue.

**Comment by bruderol** [comment]
no, sorry, @mi-we was mainly looking at the static resources (js, html, etc.) and not the REST calls, as far as I understood.

**Comment by bruderol** [comment]
@danielsuter Let's discuss tomorrow shortly whether, when and how to address this - not sure ...
